packages/store-ui/src/molecules/Carousel/hooks/useCarousel.ts

```diff
@@ -0,0 +1,177 @@
+import type { Dispatch } from 'react'
+import { useReducer } from 'react'
+import type { SwipeableProps } from 'react-swipeable'
+import { useSwipeable } from 'react-swipeable'
+
+export type SlideDirection = 'next' | 'previous'
+
+interface NextPageAction {
+  type: 'NEXT_PAGE'
+}
+
+interface PreviousPageAction {
+  type: 'PREVIOUS_PAGE'
+}
+
+interface GoToPageAction {
+  type: 'GO_TO_PAGE'
+  payload: {
+    pageIndex: number
+  }
+}
+
+interface StopSlideAction {
+  type: 'STOP_SLIDE'
+}
+
+export type Action =
+  | NextPageAction
+  | PreviousPageAction
+  | StopSlideAction
+  | GoToPageAction
+
+export type CarouselDispatch = Dispatch<Action>
+
+export interface CarouselState {
+  /**
+   * The `currentSlide` in a Carousel with multiple items in a single page is
```

Comment:
Add a @description so vscode shows a rich doc when hovering over the prop
```suggestion
   * @description The `currentSlide` in a Carousel with multiple items in a single page is
```

Reply:
Are you sure this is necessary? VSCode seems to pick up these comments even without this dierctive.

Reply:
Take care if it appears on story-ui docs